lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1603,12 +1629,13 @@ int swift::performFrontend(ArrayRef<const char *> Args,
   // arguments are generated by the driver, not directly by a user. The driver
   // is responsible for emitting diagnostics for its own errors. See SR-2683
   // for details.
+  //
+  // FIXME: Do we need one SerializedConsumer per primary? Owned by the
+  // SourceFile?
```

Comment:
I'll figure this out later, but they won't be owned by the SourceFile because they need to come into existence before the SourceFile does.

Reply:
OK